stdlib/test/builtin/test_bfloat16.mojo

```diff
@@ -35,46 +34,58 @@ def test_methods():
     assert_almost_equal(BFloat16(2.0), 2.0)
 
 
-fn test_bf_primitives():
+def test_bf_primitives():
     # we have to use dynamic values, otherwise these get evaled at compile time.
     var a = randn_float64().cast[DType.bfloat16]()
     var b = randn_float64().cast[DType.bfloat16]()
 
-    print(a + b)
-    print(a - b)
-    print(a / b)
-    print(a * b)
-    print(a == b)
-    print(a != b)
-    print(a <= b)
-    print(a >= b)
+    # higher precision
+    var a_hp = a.cast[DType.float64]()
+    var b_hp = b.cast[DType.float64]()
 
+    assert_almost_equal(a + b, (a_hp + b_hp).cast[DType.bfloat16]())
+    assert_almost_equal(a - b, (a_hp - b_hp).cast[DType.bfloat16]())
+    assert_almost_equal(a / b, (a_hp / b_hp).cast[DType.bfloat16]())
+    assert_almost_equal(a * b, (a_hp * b_hp).cast[DType.bfloat16]())
+    assert_equal(a == b, a_hp == b_hp)
+    assert_equal(a != b, a_hp != b_hp)
+    assert_equal(a <= b, a_hp <= b_hp)
+    assert_equal(a >= b, a_hp >= b_hp)
 
-def main():
-    # CHECK: 33.0
-    print(
+
+def check_float64_values():
```

Comment:
Done here as well

Reply:
Using `assert_*` function means that the function can now raise. So we have two choices, either we use `raising` to the function, either we use `def` instead of `fn`. I chose def, but I don't have a strong opinion about this